lib/formz.dart

```diff
@@ -94,7 +94,18 @@ abstract class FormzInput<T, E> {
 
   /// Returns a validation error if the [FormzInput] is invalid.
   /// Returns `null` if the [FormzInput] is valid.
-  E? get error => validator(value);
+  E? get error => _checkResultsMap();
+
+  E? _checkResultsMap() {
+    E? error;
+    if (!_forms<T, E>().resMap.containsKey(this)) {
+      error = validator(value);
+      _forms<T, E>().resMap[this] = error;
+    } else {
+      error = _forms<T, E>().resMap[this];
+    }
+    return error;
+  }
```

Comment:
This won't work and we may have a memory leak. Formz inputs are immutable, making instances of inputs short lived. Every value change creates a new input instance. 


As a result, `this` cannot be reliable as a key for this global registry.